### ISSUE TITLE
"Remember my username" does not work in simplesamlphp2.0.2.

### ISSUE THREAD

**Issue by sinbino** [simplesamlphp/simplesamlphp#1796]
# **Help us to keep this issue-tracker clean!  For questions or support, please refer to our [mailing lists](https://simplesamlphp.org/lists)**

**Describe the bug**
"Remember my username" does not work in simplesamlphp2.0.2.

**To Reproduce**
1. go to the simplesamlphp login screen.
2. Enter your user name.
3. Check "Remember my username".
4. Press the login button.
5. When the login screen is displayed again, the user name is cleared.

**Expected behavior**
If the login screen appears again, the previously entered user name is pre-populated.

**Screenshots or logs**
N/A

**Additional context**
simplesamlphp is trying to set the username cookie in the response header when posting the login screen form, but it is failing.

My browser (Chrome/111.0.5563.111) is displaying the following error in set-cookie.
````This attempt to set a cookie via a Set-Cookie header was blocked because it had the "SameSite=None" attribute but did not have the "Secure" attribute, which is required in order to use "SameSite=None".````

In previous versions of simplesamlphp (e.g. 1.19.1) the Remember username cookie was set as follows
```set-cookie: mymodule-username=blah; expires=Thu, 28-Mar-2024 03:30:02 GMT; Max-Age=31536000; path=/; secure; HttpOnly; SameSite=Lax```

In simplesamlphp 2.0.2 the Remember username cookie is set as follows
```set-cookie: mymodule-username=blah; expires=Thu, 04-May-2023 15:26:37 GMT; Max-Age=3153600; path=/; httponly; samesite=none```

The cookie with samesite=none is not accepted by the browser because security is not set, thus the "Remeber my username" is not working.
There does not seem to be a config in simplesamlphp 2.0.2 that can control the secure and samesite flags of the Remember username cookie.

**Comment by andreaswig** [comment]
It seems that in 2.0.3, SimpleSAMLphp is not even trying to set the cookie if a username or password is 
set. '$t->headers->setCookie($cookie);' in line 306 of the core-Controller 'Login.php' is not reached then.

**Comment by tvdijen** [comment]
> There does not seem to be a config in simplesamlphp 2.0.2 that can control the secure and samesite flags of the Remember username cookie.

I don't think it's desirable to add this setting for every individual cookie. You should probably configure your PHP-installation to default to `true` here.

**Comment by sinbino** [comment]
@tvdijen
Sorry it has taken so long to reply.
I have tried the patch, but it still does not seem to set the cookie.

**Comment by sinbino** [comment]
I tried to trace with v2.0.4.

modules/core/src/Controller/Login.php#136 - ENTRY Login::handleLogin
modules/core/src/Controller/Login.php#226 - CALL UserPassBase::handleLogin
modules/core/src/Auth/UserPassBase.php#284 - ENTRY UserPassBase::handleLogin
modules/core/src/Auth/UserPassBase.php#317 - CALL Auth\Source::completeAuth 

completeAuth function throws an exception and NEVER RETURNS

The code where the cookie is set is after modules/core/src/Controller/Login.php#226 and appears to never be reached if the login is successful.

**Comment by MaximilianKresse** [comment]
I stumbled over this bug too in SimpleSAML 2.1.0. 

> > There does not seem to be a config in simplesamlphp 2.0.2 that can control the secure and samesite flags of the Remember username cookie.
> 
> I don't think it's desirable to add this setting for every individual cookie. You should probably configure your PHP-installation to default to `true` here.
> 
> _edit_ : Could you try this [patch ](https://github.com/simplesamlphp/simplesamlphp/commit/d562178d30c99c7d6454cb9b4bde7b0fd4ab43ee)to see if it changes anything for you?

This patch fixes the problem for me. The main problem is clearly that you set a SameSite 'none' cookie (on some user agents) without setting it to secure. This fix should go back into the main code - I'm a little bit shocked that it wasn't yet even if this bug report is already 6 months old. 
But maybe @sinbino has another bug on top of it.

**Comment by tvdijen** [comment]
Correct, it is impossible to fix this right now without breaking backwards compatibility.
It was fixed in `master`, but that's really not an option to use due to major redesign / work-in-progress
